subsys/mgmt/mcumgr/transport/src/smp.c

```diff
@@ -131,9 +141,16 @@ smp_handle_reqs(struct k_work *work)
 
 	smpt = (void *)work;
 
+	/* Read and handle received messages  */
 	while ((nb = net_buf_get(&smpt->fifo, K_NO_WAIT)) != NULL) {
 		smp_process_packet(smpt, nb);
 	}
+#ifdef CONFIG_SMP_CLIENT
+	/* Send Client request  */
+	while ((nb = net_buf_get(&smpt->tx_fifo, K_NO_WAIT)) != NULL) {
+		smpt->functions.output(nb);
+	}
```

Comment:
Then we have one object linked list what transport layer poll and use.

Reply:
Here we need to balance waste of ram or extend code base which one is important.

Reply:
I'll have a think about how this can be done, might just need to use a bitmask field to specify which modes are enabled, 1 bit for server, 1 bit for client.

As an aside: have you tested using e.g. the UART transport in both server and client modes at the same time? I like the idea that both can be used, but would think that client would need to take priority and block server mode if it is awaiting for something to be returned.

Reply:
I think if I add tx fifo to SMP client structure and use only pointer at smp transport structure that is balanced solution. SMP client init will set pointer and init list.

Reply:
Server and client work fine same time.  Client is blocking request when wait a response but server can be run background. Now transport tx and rx is handled at same worker queue.